feature_engine/datetime/date.py

```diff
@@ -0,0 +1,169 @@
+# Authors: dodoarg <eargiolas96@gmail.com>
+
+from typing import List, Optional, Union
+
+import numpy as np
+import pandas as pd
+
+from feature_engine.dataframe_checks import _is_dataframe
+from feature_engine.datetime.base_transformer import DateTimeBaseTransformer
+from feature_engine.variable_manipulation import _check_input_parameter_variables
+
+
+class ExtractDateFeatures(DateTimeBaseTransformer):
+    """
+    ExtractDateFeatures extract various date-related features from
+    datetime features present in the dataset, adding new columns
+    accordingly.
+
+    **Notes**
+    Day of the month extraction isn't supported by panda modules.
+    This transformer implements a workaround that counts how many blocks
+    of 7 days have passed in a given month.
+    A more serious approach based on the calendar should be implemented
+    ***
+
+    Parameters
+    ----------
+    variables: list, default=None
+        The list of variables to impute. If None, the imputer will find and
+        select all datetime variables, including those converted from
+        features of type object/category.
+
+    features_to_extract: list, default = "year"
+        The list of date features to extract. See supported attribute
+
+    drop_datetime: bool, default = "True"
+        Whether to drop datetime variables from the dataframe, including
+        those that have been converted by the transformer.
+        Note: if you pass a subset of features via the variables argument
+        of the transformer and drop_datetime=True, there might be more
+        datetime features in the dataframe that will not be dropped
+        upon calling the transform method
+
+    kwargs:
+        see pd.to_datetime keyword arguments other than "errors"
+
+    Attributes
+    ----------
+    supported:
+        list of supported date features
+
+    Available upon calling the fit method:
+
+    variables_:
+        List of variables from which to extract date features
+
+    n_features_in_:
+        The number of features in the train set used in fit.
+
+    Methods
+    -------
+    fit:
+        Find the variables from which to extract date features
+    transform:
+        Add the new date features specified in features_to_extract argument
+    fit_transform:
+        Fit to the data, then transform it.
+    """
+
+    def __init__(
+        self,
+        variables: Union[None, int, str, List[Union[str, int]]] = None,
+        features_to_extract: Union[str, List[str]] = "year",
+        drop_datetime: bool = True,
+        **kwargs
```

Comment:
what would be the kwargs for?

In general, unless there is a very good reason, I would not allow the use of kwargs. As a user I almost never really know what a kwarg could be when i see them in the docs, lol

Reply:
The kwargs were meant to be propagated up to the pd.to_datetime call. The thing is, when parsing datetime data there's a couple things to consider, most importantly, the format - e.g. d/m/y or y/m/d - and pd.to_datetime has options to deal with that. <br> Currently, the transformer will always return extracted features from datetime that was converted with the default options, and the user won't have a say in that

Reply:
can we instead specifically pass the pandas dt parameters in the init, those that we would need? like coerce and the string format? or are they too many parameters? sorry I ask, I don't remember of the top of my head.

[here](https://github.com/feature-engine/feature_engine/blob/main/feature_engine/selection/drop_correlated_features.py#L89) for example the param method is the param taken by pandas corr.

Reply:
That could work. To keep things simple, I think we can get away with dayfirst and yearfirst only ([see here](https://pandas.pydata.org/docs/reference/api/pandas.to_datetime.html?highlight=pd%20to_datetime#:~:text=return%20the%20input.-,dayfirstbool%2C%20default%20False,first%20(this%20is%20a%20known%20bug%2C%20based%20on%20dateutil%20behavior).,-utcbool%2C%20default%20None)) which are obviously clutch to extract the features correctly